sdk/core/System.ClientModel/tests/client/Internal/ModelReaderWriterExtensions.cs

```diff
@@ -0,0 +1,264 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.ClientModel.Primitives;
+using System.Text.Json;
+
+namespace System.ClientModel.Tests.Client
+{
+    internal static class ModelReaderWriterExtensions
+    {
+        // TODO: These are copied from shared source files. If they become
+        // public we need to refactor and consolidate to a single place.
+
+        #region JsonElement
+
+        public static object? GetObject(in this JsonElement element)
+        {
+            switch (element.ValueKind)
+            {
+                case JsonValueKind.String:
+                    return element.GetString();
+                case JsonValueKind.Number:
+                    if (element.TryGetInt32(out int intValue))
+                    {
+                        return intValue;
+                    }
+                    if (element.TryGetInt64(out long longValue))
+                    {
+                        return longValue;
+                    }
+                    return element.GetDouble();
+                case JsonValueKind.True:
+                    return true;
+                case JsonValueKind.False:
+                    return false;
+                case JsonValueKind.Undefined:
```

Comment:
when do we get this? should we throw?

Reply:
This is not new or added with MRW.  This is the existing code originally called Utf8JsonWriterExtensions.  I wouldn't want to touch this as it could affect existing behaviors.